yt_dlp/extractor/mxplayer.py

```diff
@@ -106,73 +108,99 @@ class MxplayerIE(InfoExtractor):
+            'series': 'Ek Thi Begum (Hindi)',
+            'season': 'Season 2',
+            'season_number': 2,
+            'episode': 'Episode 2',
+            'episode_number': 2,
+        },
+        'params': {
+            'format': 'bv',
+            'skip_download': True,
+        },
     }]
 
     def _real_extract(self, url):
-        type, display_id, video_id = self._match_valid_url(url).groups()
-        type = 'movie_film' if type == 'movie' else 'tvshow_episode'
-        API_URL = 'https://androidapi.mxplay.com/v1/detail/'
-        headers = {
-            'X-Av-Code': '23',
-            'X-Country': 'IN',
-            'X-Platform': 'android',
-            'X-App-Version': '1370001318',
-            'X-Resolution': '3840x2160',
-        }
-        data_json = self._download_json(f'{API_URL}{type}/{video_id}', display_id, headers=headers)['profile']
-
-        season, series = None, None
-        for dct in data_json.get('levelInfos', []):
-            if dct.get('type') == 'tvshow_season':
-                season = dct.get('name')
-            elif dct.get('type') == 'tvshow_show':
-                series = dct.get('name')
-        thumbnails = []
-        for thumb in data_json.get('poster', []):
-            thumbnails.append({
-                'url': thumb.get('url'),
-                'width': thumb.get('width'),
-                'height': thumb.get('height'),
-            })
+        video_type, display_id, video_id = self._match_valid_url(url).groups()
+
+        if 'show' in video_type:
+            video_type = 'episode'
+
+        data_json = self._download_json(
+            f'https://api.mxplay.com/v1/web/detail/video?type={video_type}&id={video_id}', display_id
+        )
 
         formats = []
         subtitles = {}
-        for dct in data_json.get('playInfo', []):
-            if dct.get('extension') == 'mpd':
-                frmt, subs = self._extract_mpd_formats_and_subtitles(dct.get('playUrl'), display_id, fatal=False)
-                formats.extend(frmt)
-                subtitles = self._merge_subtitles(subtitles, subs)
-            elif dct.get('extension') == 'm3u8':
-                frmt, subs = self._extract_m3u8_formats_and_subtitles(dct.get('playUrl'), display_id, fatal=False)
-                formats.extend(frmt)
-                subtitles = self._merge_subtitles(subtitles, subs)
+        series, season, season_number, episode_number = None, None, None, None
+
+        if video_type == 'episode':
+            series = traverse_obj(data_json, ('container', 'container', 'title'))
+            season = traverse_obj(data_json, ('container', 'title'))
+            episode_number = data_json.get('sequence')
+            season_number = int(self._search_regex(r'Season (\d+)', season, 'Season Number'))
```

Comment:
By the way, can this be inline? If the values don't exist, we'll get None by default

Reply:
This is done. Had to use an if-else for avoiding return of 0 as episode number and instead return None.